rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -85,42 +85,45 @@ type EconomicTerms: <" This class represents the full set of price-forming featu
         if payout -> assetPayout is absent
         then // Not Security Lending/Repo so must be Interest Rate Swap, which must set an exercise option type
             if terminationProvision -> extendibleProvision exists
-            then (terminationProvision -> extendibleProvision -> americanExercise exists
-                        and terminationProvision -> extendibleProvision -> bermudaExercise is absent
-                        and terminationProvision -> extendibleProvision -> europeanExercise is absent)
-                    or (terminationProvision -> extendibleProvision -> americanExercise is absent
-                        and terminationProvision -> extendibleProvision -> bermudaExercise exists
-                        and terminationProvision -> extendibleProvision -> europeanExercise is absent)
-                    or (terminationProvision -> extendibleProvision -> americanExercise is absent
-                        and terminationProvision -> extendibleProvision -> bermudaExercise is absent
-                        and terminationProvision -> extendibleProvision -> europeanExercise exists)
+            then (terminationProvision -> extendibleProvision -> exerciseTerms -> style = OptionExerciseStyleEnum -> American 
+                        and terminationProvision -> extendibleProvision -> exerciseTerms -> style <> OptionExerciseStyleEnum -> Bermuda
+                        and terminationProvision -> extendibleProvision -> exerciseTerms -> style <> OptionExerciseStyleEnum -> European)
+                    or (terminationProvision -> extendibleProvision -> exerciseTerms -> style <> OptionExerciseStyleEnum -> American
+                        and terminationProvision -> extendibleProvision -> exerciseTerms -> style = OptionExerciseStyleEnum -> Bermuda
+                        and terminationProvision -> extendibleProvision -> exerciseTerms -> style <> OptionExerciseStyleEnum -> European)
+                    or (terminationProvision -> extendibleProvision -> exerciseTerms -> style <> OptionExerciseStyleEnum -> American
+                        and terminationProvision -> extendibleProvision -> exerciseTerms -> style <> OptionExerciseStyleEnum -> Bermuda
+                        and terminationProvision -> extendibleProvision -> exerciseTerms -> style = OptionExerciseStyleEnum -> European)
                         and terminationProvision -> extendibleProvision -> followUpConfirmation exists
 
+
 type OptionExercise: <" A class to represent the applicable terms to qualify an option exercise: the option style (e.g. American style option), the exercise procedure (e.g. manual exercise) and the settlement terms (e.g. physical vs. cash).">
+    [deprecated]
```

Comment:
Remove type altogether. Backward-incompatible changes are allowed in the dev version.